impl/src/main/kotlin/nl/nuts/consent/bridge/listener/StateFileStorage.kt

```diff
@@ -0,0 +1,155 @@
+/*
+ * Nuts consent bridge
+ * Copyright (C) 2020 Nuts community
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package nl.nuts.consent.bridge.listener
+
+import nl.nuts.consent.bridge.EventMetaProperties
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Service
+import java.io.EOFException
+import java.io.File
+import java.io.RandomAccessFile
+import java.nio.channels.FileChannel
+import java.nio.channels.FileLock
+
+private const val FILE_EXTENSION = "stamp"
+
+/**
+ * Controls access to different files used for storing the latest event timestamp of Corda events
+ */
+@Service
+class StateFileStorageControl {
+
+    @Autowired
+    lateinit var eventMetaProperties: EventMetaProperties
+
+    val stateFileStorages: MutableMap<String, StateFileStorage> = mutableMapOf()
```

Comment:
we couldn't cluster Corda anyways, right?

Reply:
Nope, that's a whole different ball game.